src/Compilers/Core/MSBuildTask/Csc.cs

```diff
@@ -200,10 +201,19 @@ protected override string ToolNameWithoutExtension
         /// </summary>
         protected override void AddResponseFileCommands(CommandLineBuilderExtension commandLine)
         {
-            // Pass sdkpath if we are invoking core compiler from framework to preserve the behavior that framework compiler would have.
+            // Pass sdkpath and csc.rsp if we are invoking core compiler from framework to preserve the behavior that framework compiler would have.
             if (IsSdkFrameworkToCoreBridgeTask)
             {
                 commandLine.AppendSwitchIfNotNull("/sdkpath:", RuntimeEnvironment.GetRuntimeDirectory());
+
+                if (!NoConfig)
+                {
+                    var rspFile = Path.Combine(Path.GetDirectoryName(typeof(ManagedCompiler).Assembly.Location)!, "csc.rsp");
+                    if (File.Exists(rspFile))
+                    {
+                        commandLine.AppendSwitchIfNotNull("@", rspFile);
```

Comment:
>"@" [](http://example.com/codeflow?start=58&length=3)

It doesn't look like we are testing scenario when an .rsp file is already specified